src/Lib.P2P.Tests/PeerManagerTest.cs

```diff
@@ -0,0 +1,173 @@
+            PublicKey =
+                "CAASXjBcMA0GCSqGSIb3DQEBAQUAA0sAMEgCQQCC5r4nQBtnd9qgjnG8fBN5+gnqIeWEIcUFUdCG4su/vrbQ1py8XGKNUBuDjkyTv25Gd3hlrtNJV3eOKZVSL8ePAgMBAAE="
+        };
+
+        [TestMethod]
+        public void IsNotReachable()
+        {
+            var peer = new Peer {Id = "QmXFX2P5ammdmXQgfqGkfswtEVFsZUJ5KeHRXQYCTdiTAb"};
+            var manager = new PeerManager {SwarmService = new SwarmService()};
+            Assert.AreEqual(0, manager.DeadPeers.Count);
+
+            manager.SetNotReachable(peer);
+            Assert.IsTrue(manager.DeadPeers.ContainsKey(peer));
+            Assert.AreEqual(1, manager.DeadPeers.Count);
+
+            manager.SetNotReachable(peer);
+            Assert.IsTrue(manager.DeadPeers.ContainsKey(peer));
+            Assert.AreEqual(1, manager.DeadPeers.Count);
+
+            manager.SetReachable(peer);
+            Assert.IsFalse(manager.DeadPeers.ContainsKey(peer));
+            Assert.AreEqual(0, manager.DeadPeers.Count);
+        }
+
+        [TestMethod]
+        public void BlackListsThePeer()
+        {
+            var peer = new Peer {Id = "QmXFX2P5ammdmXQgfqGkfswtEVFsZUJ5KeHRXQYCTdiTAb"};
+            var manager = new PeerManager {SwarmService = new SwarmService()};
+            Assert.AreEqual(0, manager.DeadPeers.Count);
+
+            manager.SetNotReachable(peer);
+            Assert.IsFalse(
+                manager.SwarmService.IsAllowed((MultiAddress) "/p2p/QmXFX2P5ammdmXQgfqGkfswtEVFsZUJ5KeHRXQYCTdiTAb"));
+
+            manager.SetReachable(peer);
+            Assert.IsTrue(
+                manager.SwarmService.IsAllowed((MultiAddress) "/p2p/QmXFX2P5ammdmXQgfqGkfswtEVFsZUJ5KeHRXQYCTdiTAb"));
+        }
+
+        [TestMethod]
+        public async Task Backoff_Increases()
+        {
+            var peer = new Peer
+            {
+                Id = "QmXFX2P5ammdmXQgfqGkfswtEVFsZUJ5KeHRXQYCTdiTxx",
+                Addresses = new MultiAddress[]
+                {
+                    "/ip4/127.0.0.1/tcp/4040/ipfs/QmXFX2P5ammdmXQgfqGkfswtEVFsZUJ5KeHRXQYCTdiTxx"
+                }
+            };
+            var swarm = new SwarmService {LocalPeer = self};
+            var manager = new PeerManager
+            {
+                SwarmService = swarm,
+                InitialBackoff = TimeSpan.FromMilliseconds(100),
+            };
+            Assert.AreEqual(0, manager.DeadPeers.Count);
+
+            try
+            {
+                await swarm.StartAsync();
+                await manager.StartAsync();
+                try
+                {
+                    await swarm.ConnectAsync(peer);
+                }
+                catch
+                {
+                    // ignored
+                }
+
+                Assert.AreEqual(1, manager.DeadPeers.Count);
+
+                var end = DateTime.Now + TimeSpan.FromSeconds(4);
+                while (DateTime.Now <= end)
+                    if (manager.DeadPeers[peer].Backoff > manager.InitialBackoff)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)